tf2_sensor_msgs/src/tf2_sensor_msgs/tf2_sensor_msgs.py

```diff
@@ -47,24 +47,71 @@ def from_msg_msg(msg):
 tf2_ros.ConvertRegistration().add_from_msg(PointCloud2, from_msg_msg)
 
 
-def transform_to_kdl(t):
-    return PyKDL.Frame(PyKDL.Rotation.Quaternion(
-                            t.transform.rotation.x, t.transform.rotation.y,
-                            t.transform.rotation.z, t.transform.rotation.w),
-                       PyKDL.Vector(t.transform.translation.x,
-                                    t.transform.translation.y,
-                                    t.transform.translation.z))
+def transform_points(point_cloud: np.ndarray, transform: Transform) -> np.ndarray:
+    """
+    Transforms a bulk of points from an numpy array using a provided `Transform`.
+
+    :param point_cloud: nx3 Array of points where n is the number of points
+    :param transform: TF2 transform used for the transformation
+    :returns: Array with the same shape as the input array, but with the transformation applied
+    """
+    # Build affine transformation
+    transform_translation = np.array([
+        transform.translation.x,
+        transform.translation.y,
+        transform.translation.z
+    ])
+    transform_rotation_matrix = _get_mat_from_quat(
+        np.array([
+            transform.rotation.w,
+            transform.rotation.x,
+            transform.rotation.y,
+            transform.rotation.z
+        ]))
+
+    # "Batched" matmul meaning a matmul for each point
+    # First we offset all points by the translation part followed by a rotation using the rotation matrix
+    return np.einsum("ij, pj -> pi", transform_rotation_matrix, point_cloud + transform_translation)
```

Comment:
Single quotes preferred over double quotes, from the style guide: https://docs.ros.org/en/rolling/Contributing/Code-Style-Language-Versions.html#id4

```suggestion
    return np.einsum('ij, pj -> pi', transform_rotation_matrix, point_cloud + transform_translation)
```